libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -197,9 +204,27 @@ - (void)send:(SFRestRequest *)request delegate:(id<SFRestDelegate>)delegate shou
         }];
     } else {
 
-        // Auth credentials exist. Just send the request.
-        [self enqueueRequest:request delegate:delegate shouldRetry:shouldRetry];
+        /*
+         * Auth credentials exist. Send the request if an OAuth session is not being refreshed.
+         * Otherwise, wait for the current session refresh call to complete before sending.
+         */
+        if (!self.sessionRefreshInProgress) {
```

Comment:
Attempt to send the request only if a session refresh is not already in progress. Otherwise, wait for it to finish - queued requests will automatically be processed at this point.